server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationIT.java

```diff
@@ -1343,4 +1347,76 @@ public void testPrimaryReceivesDocsDuringReplicaRecovery() throws Exception {
         ensureGreen(INDEX_NAME);
         waitForSearchableDocs(2, nodes);
     }
+
+    public void testSimpleDFSQueryWithSegmentReplication() throws Exception {
```

Comment:
nit - this test is less about the dfs query and more that we are indexing while the second node is recovering.  Can we rename to reflect this? Something like `testIndexWhileRecoveringReplica`.